index.js

```diff
@@ -190,16 +228,38 @@ function redisStore(args) {
     options = options || {};
 
     var ttl = (options.ttl || options.ttl === 0) ? options.ttl : redisOptions.ttl;
+    var gzip = (options.gzip || options.gzip === false) ? options.gzip : redisOptions.gzip;
+    if (gzip === true) {
+      gzip = gzipDefault;
+    }
 
     connect(function(err, conn) {
       if (err) {
         return cb && cb(err);
       }
       var val = JSON.stringify(value) || '"undefined"';
-      if (ttl) {
-        conn.setex(key, ttl, val, handleResponse(conn, cb));
-      } else {
-        conn.set(key, val, handleResponse(conn, cb));
+
+      if (gzip) {
+        zlib.gzip(val, gzip, function (gzErr, gzVal) {
+          if (gzErr) {
+            return cb && cb(gzErr);
+          }
+
+          if (ttl) {
+            conn.setex(key, ttl, gzVal, handleResponse(conn, cb));
+          }
+          else {
+            conn.set(key, gzVal, handleResponse(conn, cb));
+          }
+        });
+      }
+      else {
+        if (ttl) {
+          conn.setex(key, ttl, val, handleResponse(conn, cb));
+        }
+        else {
+          conn.set(key, val, handleResponse(conn, cb));
```

Comment:
Yes, I agree. The `setex` should be replaced, but I'm not sure if the `node-redis` module supports the new format for the `set` command, yet. Has anyone tested that?